modules/azure/aks/aks.tf

```diff
@@ -62,6 +62,56 @@ resource "azurerm_kubernetes_cluster" "this" {
   }
 }
 
+resource "azurerm_monitor_diagnostic_setting" "this" {
+  name                       = "aks-${var.environment}-${var.location_short}-${var.name}${var.aks_name_suffix}"
+  target_resource_id         = azurerm_kubernetes_cluster.this.id
+  log_analytics_workspace_id = var.aks_log_analytics_id
+
+  log {
+    category = "kube-scheduler"
+    enabled  = true
+
+    retention_policy {
+      enabled = true
+      days    = 7
+    }
+  }
+
+  log {
+    category = "kube-controller-manager"
+    enabled  = false
+
+    retention_policy {
+      enabled = false
+    }
+  }
+
+  log {
+    category = "cluster-autoscaler"
+    enabled  = true
+
+    retention_policy {
+      enabled = true
+      days    = 7
+    }
+  }
+
+  log {
+    category = "kube-audit"
+    enabled  = true
+
+    retention_policy {
+      enabled = true
+      days    = 30
```

Comment:
The retention time for the audit log is rather low.
One option forward could be to have two azurerm_monitor_diagnostic_setting definitions.
One where we send data to log analytics and the second where we send the data to a storage account.

This would lower the cost of the aduit logs by allot. Then we would of course have to create a good way of working of easily reading out this data.

Another option could be to have a event-hub in between get all the data, send some to log analytics and some to storage account.

And a third option is to send all the data to a storage account but then we have to be really good at reading the data from the storage account.